### PR TITLE
add more details to messages logged in entity scripts

### PR DESCRIPTION
**Pull request by odysseus654**
Attempt to improve the amount of detail logged by entity scripts so we have an idea of where the messages might be coming from.

New proposal:
```
[08/30 23:46:33] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
[08/30 23:46:33] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
[08/30 23:46:33] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
[08/30 23:46:33] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
[08/30 23:46:34] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
[08/30 23:46:34] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
[08/30 23:46:34] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
[08/30 23:46:34] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
[08/30 23:46:34] [DEBUG] [hifi.scriptengine.script] [:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js] 0.00033786098356358707
```

more specifically: \<function\>:\<line number\> at \<filename\>

### CODE REVIEW

**Review by daleglass — APPROVED** [review]

**Comment by daleglass** [comment]
Code looks good, though the :45 at the beginning is a tad unusual. Maybe have it be

     [https://transmissiongate.com/hifi/assoa/sitterScript.js:45]


instead, like most compilers and such do?

**Comment by odysseus654** [comment]
My intent is for it to be something like [doSomethingFunction:45 at https://transmissiongate.com/hifi/assoa/sitterScript.js].

Doesn't really work if the function is unnamed.

**Comment by kasenvr** [comment]
Is there any way to put this improved capability behind a toggle? Because she might just end up taking way too much space in the log.

**Comment by daleglass** [comment]
@odysseus654 Is that line number within the function, or within the file? If it's within the file, I think filename:lineno is still more understandable.

@kasenvr On a longer term, I'd like to have a better logging system that's easily adjustable and logs lots of useful stuff. But I think maybe that's for later. For now I'm happy if I can see where errors are coming from.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-5958c69_ee43cfd8.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-5958c69.tar_ee43cfd8.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-5958c69.tar_ee43cfd8.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-5958c69_ee43cfd8.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-5958c69_ee43cfd8.exe

**Comment by kasenvr** [comment]
I fixed 'er right up. 
image

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-157e433_4096d75b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-157e433.tar_4096d75b.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-157e433.tar_4096d75b.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-157e433_4096d75b.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-157e433_4096d75b.exe

**Review by daleglass — APPROVED** [review]

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 1]
```diff
@@ -117,12 +118,53 @@ static QScriptValue debugPrint(QScriptContext* context, QScriptEngine* engine) {
         message += context->argument(i).toString();
     }
 
-    if (ScriptEngine *scriptEngine = qobject_cast<ScriptEngine*>(engine)) {
+    // was this generated by a script engine? If we don't recognize it then send the message and exit
+    ScriptEngine* scriptEngine = qobject_cast<ScriptEngine*>(engine);
+    if (scriptEngine == nullptr) {
```
Comment (HifiExperiments) [inline libraries/script-engine/src/ScriptEngine.cpp:123]: !scriptEngine (similarly, for some checks below, the coding standard says to prefer not writing out nullptr)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-b94191c.tar_ab46b1fc.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-b94191c.tar_ab46b1fc.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-b94191c_ab46b1fc.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-b94191c_ab46b1fc.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-b94191c_ab46b1fc.exe

**Review by HifiExperiments — APPROVED** [review]

**Comment by ctrlaltdavid** [comment]
There has been a general push to remove identifying URLs from the program log. does this run afoul of that intent?

**Comment by kasenvr** [comment]
> 
> 
> There has been a general push to remove identifying URLs from the program log. does this run afoul of that intent?

Well there was, long ago, but that was High Fidelity. That hurts things more than helps most often. Even so, this will only reveal running scripts which are also a security thing whereas avatars are less so.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-4b46a16.tar_853c8052.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-4b46a16.tar_853c8052.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-4b46a16_853c8052.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-4b46a16_853c8052.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR669/Vircadia-Alpha-PR669-4b46a16_853c8052.exe

**Comment by kasenvr** [comment]
Given that this is not dropping performance, I have tested it and confirmed it to work in general.

**Comment by kasenvr** [comment]
Given there are no performance concerns from CR, I'm OK with considering this QA approved.

**Pull request merged**